Docs need an example to show use of a rerank driver with RAG engines

The sample here does not show how to use a rerank driver with a RAG Engine

https://docs.griptape.ai/stable/griptape-framework/engines/rag-engines/

```
docs/griptape-framework/engines/src/rag_engines_1.py:1:1: I001 [*] Import block is un-sorted or un-formatted
   |
 1 | / from griptape.chunkers import TextChunker
 2 | | from griptape.drivers.embedding.openai import OpenAiEmbeddingDriver
 3 | | from griptape.drivers.prompt.openai import OpenAiChatPromptDriver
 4 | | from griptape.drivers.vector.local import LocalVectorStoreDriver
 5 | | from griptape.drivers.rerank.local import LocalRerankDriver
 6 | | from griptape.engines.rag import RagContext, RagEngine
 7 | | from griptape.engines.rag.modules import (
 8 | |     PromptResponseRagModule,
 9 | |     TranslateQueryRagModule,
10 | |     VectorStoreRetrievalRagModule,
11 | |     TextChunksRerankRagModule,
12 | | )
13 | | from griptape.engines.rag.stages import (
14 | |     QueryRagStage,
15 | |     ResponseRagStage,
16 | |     RetrievalRagStage,
17 | | )
18 | | from griptape.loaders import WebLoader
19 | | from griptape.rules import Rule, Ruleset
   | |________________________________________^ I001
20 |
21 |   prompt_driver = OpenAiChatPromptDriver(model="gpt-4o", temperature=0)
   |
   = help: Organize imports

Found 1 error.
[*] 1 fixable with the `--fix` option.
make: *** [check/lint] Error 1
```

--- a/docs/griptape-framework/engines/rag-engines.md
+++ b/docs/griptape-framework/engines/rag-engines.md
@@ -21,19 +21,19 @@ search:
 
 ### RAG Modules
 
-RAG modules are used to implement concrete actions in the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies.
+RAG modules are used to implement concrete actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. The stages of the pipeline and associated modules are as follows:
 
-#### Query Modules
+#### Query RAG Stage
 
 - `TranslateQueryRagModule` is for translating the query into another language.
 
-#### Retrieval/Rerank Modules
+#### Retrieval RAG Stage
 
 - `TextChunksRerankRagModule` is for re-ranking retrieved results.
 - `TextLoaderRetrievalRagModule` is for retrieving data with text loaders in real time.
 - `VectorStoreRetrievalRagModule` is for retrieving text chunks from a vector store.
 
-#### Response Modules
+#### Response RAG Stage
 
 - `PromptResponseRagModule` is for generating responses based on retrieved text chunks.
 - `TextChunksResponseRagModule` is for responding with retrieved text chunks.
@@ -45,7 +45,7 @@ RAG modules are used to implement concrete actions in the RAG pipeline. `RagEngi
 
 ### Example
 
-The following example shows a simple RAG pipeline that translates incoming queries into English, retrieves data from a local vector store, and generates a response:
+The following example shows a simple RAG pipeline that translates incoming queries into English, retrieves data from a local vector store, reranks the results using the local rerank driver, and generates a response:
 
 ```python
 --8<-- "docs/griptape-framework/engines/src/rag_engines_1.py"

--- a/docs/griptape-framework/engines/rag-engines.md
+++ b/docs/griptape-framework/engines/rag-engines.md
@@ -17,31 +17,33 @@ search:
 
 ### RAG Modules
 
-RAG modules are used to implement actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. The stages of the pipeline, together with their purposes and associated modules, are as follows:
+RAG modules are used to implement actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. 
 
-#### Query Stage
+The three stages of the pipeline implemented in RAG Engines, together with their purposes and associated modules, are as follows:
+
+### Query Stage
 
 This stage is used for modifying input queries before they are submitted.
 
-**Query Modules**
+#### Query Stage Modules
 
 - `TranslateQueryRagModule` is for translating the query into another language.
 
-#### Retrieval Stage
+### Retrieval Stage
 
 Results are retrieved in this stage, either from a vector store in the form of chunks, or with a text loader. You may optionally use a rerank module in this stage to rerank results in order of their relevance to the original query.  
 
-**Retrieval Modules**
+#### Retrieval Stage Modules
 
 - `TextChunksRerankRagModule` is for re-ranking retrieved results.
 - `TextLoaderRetrievalRagModule` is for retrieving data with text loaders in real time.
 - `VectorStoreRetrievalRagModule` is for retrieving text chunks from a vector store.
 
-#### Response Stage
+### Response Stage
 
 Responses are generated in this final stage of the RAG pipeline. 
 
-**Response Modules**
+#### Response Stage Modules
 
 - `PromptResponseRagModule` is for generating responses based on retrieved text chunks.
 - `TextChunksResponseRagModule` is for responding with retrieved text chunks.

--- a/docs/griptape-framework/engines/rag-engines.md
+++ b/docs/griptape-framework/engines/rag-engines.md
@@ -23,17 +23,29 @@ search:
 
 RAG modules are used to implement concrete actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. The stages of the pipeline and associated modules are as follows:
 
-#### Query RAG Stage
+#### Query Stage
+
+This stage is used for modifying input queries before they are submitted.
+
+**Query Modules**
 
 - `TranslateQueryRagModule` is for translating the query into another language.
 
-#### Retrieval RAG Stage
+#### Retrieval Stage
+
+Results are retrieved in this stage, either from a vector store in the form of chunks, or with a text loader. You may optionally use a rerank module in this stage to rerank results in order of their relevance to the original query.  
+
+**Retrieval Modules**
 
 - `TextChunksRerankRagModule` is for re-ranking retrieved results.
 - `TextLoaderRetrievalRagModule` is for retrieving data with text loaders in real time.
 - `VectorStoreRetrievalRagModule` is for retrieving text chunks from a vector store.
 
-#### Response RAG Stage
+#### Response Stage
+
+Responses are generated in this final stage of the RAG pipeline. 
+
+**Response Modules**
 
 - `PromptResponseRagModule` is for generating responses based on retrieved text chunks.
 - `TextChunksResponseRagModule` is for responding with retrieved text chunks.

--- a/docs/griptape-framework/engines/rag-engines.md
+++ b/docs/griptape-framework/engines/rag-engines.md
@@ -15,25 +15,33 @@ search:
 
 `RagEngine`s consist of three _stages_: `QueryRagStage`, `RetrievalRagStage`, and `ResponseRagStage`. These stages are always executed sequentially. Each stage comprises multiple _modules_, which are executed in a customized manner. Due to this unique structure, `RagEngines` are not intended to replace [Workflows](../structures/workflows.md) or [Pipelines](../structures/pipelines.md).
 
-- `QueryRagStage` is used for modifying user queries.
-- `RetrievalRagStage` is used for retrieving and re-ranking text chunks.
-- `ResponseRagStage` is used for generating responses.
-
 ### RAG Modules
 
-RAG modules are used to implement concrete actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. The stages of the pipeline and associated modules are as follows:
+RAG modules are used to implement actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. The stages of the pipeline, together with their purposes and associated modules, are as follows:
 
-#### Query RAG Stage
+#### Query Stage
+
+This stage is used for modifying input queries before they are submitted.
+
+**Query Modules**
 
 - `TranslateQueryRagModule` is for translating the query into another language.
 
-#### Retrieval RAG Stage
+#### Retrieval Stage
+
+Results are retrieved in this stage, either from a vector store in the form of chunks, or with a text loader. You may optionally use a rerank module in this stage to rerank results in order of their relevance to the original query.  
+
+**Retrieval Modules**
 
 - `TextChunksRerankRagModule` is for re-ranking retrieved results.
 - `TextLoaderRetrievalRagModule` is for retrieving data with text loaders in real time.
 - `VectorStoreRetrievalRagModule` is for retrieving text chunks from a vector store.
 
-#### Response RAG Stage
+#### Response Stage
+
+Responses are generated in this final stage of the RAG pipeline. 
+
+**Response Modules**
 
 - `PromptResponseRagModule` is for generating responses based on retrieved text chunks.
 - `TextChunksResponseRagModule` is for responding with retrieved text chunks.

--- a/docs/griptape-framework/engines/rag-engines.md
+++ b/docs/griptape-framework/engines/rag-engines.md
@@ -17,31 +17,33 @@ search:
 
 ### RAG Modules
 
-RAG modules are used to implement actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. The stages of the pipeline, together with their purposes and associated modules, are as follows:
+RAG modules are used to implement actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. 
 
-#### Query Stage
+The three stages of the pipeline implemented in RAG Engines, together with their purposes and associated modules, are as follows:
+
+### Query Stage
 
 This stage is used for modifying input queries before they are submitted.
 
-**Query Modules**
+#### Query Stage Modules
 
 - `TranslateQueryRagModule` is for translating the query into another language.
 
-#### Retrieval Stage
+### Retrieval Stage
 
 Results are retrieved in this stage, either from a vector store in the form of chunks, or with a text loader. You may optionally use a rerank module in this stage to rerank results in order of their relevance to the original query.  
 
-**Retrieval Modules**
+#### Retrieval Stage Modules
 
 - `TextChunksRerankRagModule` is for re-ranking retrieved results.
 - `TextLoaderRetrievalRagModule` is for retrieving data with text loaders in real time.
 - `VectorStoreRetrievalRagModule` is for retrieving text chunks from a vector store.
 
-#### Response Stage
+### Response Stage
 
-Responses are generated in this final stage of the RAG pipeline. 
+Responses are generated in this final stage. 
 
-**Response Modules**
+#### Response Stage Modules
 
 - `PromptResponseRagModule` is for generating responses based on retrieved text chunks.
 - `TextChunksResponseRagModule` is for responding with retrieved text chunks.

--- a/docs/griptape-framework/engines/rag-engines.md
+++ b/docs/griptape-framework/engines/rag-engines.md
@@ -17,7 +17,7 @@ search:
 
 ### RAG Modules
 
-RAG modules are used to implement actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies. 
+RAG modules are used to implement actions in the different stages of the RAG pipeline. `RagEngine` enables developers to easily add new modules to experiment with novel RAG strategies.
 
 The three stages of the pipeline implemented in RAG Engines, together with their purposes and associated modules, are as follows:
 
@@ -31,7 +31,7 @@ This stage is used for modifying input queries before they are submitted.
 
 ### Retrieval Stage
 
-Results are retrieved in this stage, either from a vector store in the form of chunks, or with a text loader. You may optionally use a rerank module in this stage to rerank results in order of their relevance to the original query.  
+Results are retrieved in this stage, either from a vector store in the form of chunks, or with a text loader. You may optionally use a rerank module in this stage to rerank results in order of their relevance to the original query.
 
 #### Retrieval Stage Modules
 
@@ -41,7 +41,7 @@ Results are retrieved in this stage, either from a vector store in the form of c
 
 ### Response Stage
 
-Responses are generated in this final stage. 
+Responses are generated in this final stage.
 
 #### Response Stage Modules
 

--- a/docs/griptape-framework/engines/src/rag_engines_1.py
+++ b/docs/griptape-framework/engines/src/rag_engines_1.py
@@ -1,10 +1,20 @@
 from griptape.chunkers import TextChunker
 from griptape.drivers.embedding.openai import OpenAiEmbeddingDriver
 from griptape.drivers.prompt.openai import OpenAiChatPromptDriver
+from griptape.drivers.rerank.local import LocalRerankDriver
 from griptape.drivers.vector.local import LocalVectorStoreDriver
 from griptape.engines.rag import RagContext, RagEngine
-from griptape.engines.rag.modules import PromptResponseRagModule, TranslateQueryRagModule, VectorStoreRetrievalRagModule
-from griptape.engines.rag.stages import QueryRagStage, ResponseRagStage, RetrievalRagStage
+from griptape.engines.rag.modules import (
+    PromptResponseRagModule,
+    TextChunksRerankRagModule,
+    TranslateQueryRagModule,
+    VectorStoreRetrievalRagModule,
+)
+from griptape.engines.rag.stages import (
+    QueryRagStage,
+    ResponseRagStage,
+    RetrievalRagStage,
+)
 from griptape.loaders import WebLoader
 from griptape.rules import Rule, Ruleset
 
@@ -26,14 +36,18 @@ rag_engine = RagEngine(
         max_chunks=5,
         retrieval_modules=[
             VectorStoreRetrievalRagModule(
-                name="MyAwesomeRetriever", vector_store_driver=vector_store, query_params={"top_n": 20}
+                name="MyAwesomeRetriever",
+                vector_store_driver=vector_store,
+                query_params={"top_n": 20},
             )
         ],
+        rerank_module=TextChunksRerankRagModule(rerank_driver=LocalRerankDriver()),
     ),
     response_stage=ResponseRagStage(
         response_modules=[
             PromptResponseRagModule(
-                prompt_driver=prompt_driver, rulesets=[Ruleset(name="persona", rules=[Rule("Talk like a pirate")])]
+                prompt_driver=prompt_driver,
+                rulesets=[Ruleset(name="persona", rules=[Rule("Talk like a pirate")])],
             )
         ]
     ),

